docs/release_guidelines.md

```diff
@@ -1,23 +1,115 @@
+
+```
+git fetch upstream
+git checkout release-4.12
+```
+
+Note that `upstream` is the name of the remote for the base repository, not your personal fork.
+
+### Jenkins
+
+From Jenkins we will automatically be using the release branch if no other ocs-ci branch has been specified. Which specific release branch we use will be dynamically based on the `OCS_VERSION` or `UPGRADE_OCS_VERSION` parameters. For example, if `OCS_VERSION` is 4.13 we will use `release-4.13`. If this release branch doesn't exist yet, we will default back to the `stable` branch. This logic can be overridden by specifying a branch to use.
+
+## Creating Release Branches
+
+Release branches will branch off of the `stable` branch. Before we can create a release branch, we will need to know which commit to branch off from. Once we know the commit we are basing the branch off of, we can then go about creating the branch from either the git CLI or the Github UI.
+
+### Github UI
+
+From the Github UI, navigate to the list of [branches](https://github.com/red-hat-storage/ocs-ci/branches). From here you can click on new branch, selecting the `stable` branch as the base. Note that the branch creation process only works this way if the latest stable commit is the point we wish to branch off from. Otherwise we will need to create the branch using the CLI.
+
+### CLI
+
+First, ensure your local repository is up to date with the remote. Note that the following command is destructive so make sure you have any local changes stashed or committed before continuing.
+```
+git checkout master && git fetch --all && git reset --hard upstream/master
+```
+
+Checkout the stable branch
+```
+git checkout stable
+```
+
+Determine the starting point of the new release branch, for this you will need the git hash. You can find this using:
+```
+git log
+```
+
+Once you know the hash, we will create the new release branch using the hash from earlier as the base. Note that x.y is the major.minor version.
+```
+git checkout -b release-x.y 46568b9b4b8c2ceabad665b839655e0b19ab3634
+```
+
+Finally, push the branch to the remote.
+```
+git push upstream release-x.y
+```
+
+
+## Backporting Changes
+
+Most changes to the repository will end up being merged to master, propagage to the stable branch and evnetually end up on the next release branch. Critical bug fixes or important test cases may be selected to be backported to existing release branches. In order to backport changes to an existing release branch, we will need to take the following steps.
+
+### Creating a cherry-pick PR
+
+The first step in the process will be to checkout the master branch and ensure it is up to date, as well as fetch remote branches. Note that the following command is destructive so make sure you have any local changes stashed or committed before continuing.
```

Comment:
```suggestion
The first step in the process will be to checkout the master branch and ensure it is up to date, as well as fetch remote branches. 

> Note that the following command is destructive so make sure you have any local changes stashed or committed before continuing.
```

This will be better visible in the doc.